### PR TITLE
transfer ownership creates a collaborator record for the previous owner.

### LINKED ISSUE #1035: Transferring ownership should not revoke all access to a cookbook
Instead, perhaps after you transfer ownership of a cookbook, you are automatically added as a collaborator?

### PR DESCRIPTION
**Pull request by kirtfitzpatrick**
fixes #1035

### CODE REVIEW

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.chef.io/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## Non-GitHub Verified Committers

There are 1 commit author(s) whose commits are authored by a non-GitHub verified email address. Chef will have to manually verify that they are authorized to contribute.

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by nellshamrell** [comment]
:+1: When Travis tests pass

**Comment by smith** [comment]
Looks like you got dinged by the cops: https://travis-ci.org/chef/supermarket/builds/61692003#L688

File: spec/models/cookbook_spec.rb [hunk 1]
```diff
@@ -163,6 +163,17 @@
       expect(cookbook.collaborator_users).to_not include(hank)
     end
 
+    it 'should create a new collaborator record for the previous owner' do
+      hank = create(:user)
+      create(:cookbook_collaborator, resourceable: cookbook, user: hank)
+      expect(cookbook.owner).to eql(jimmy)
```
Comment (smith) [inline spec/models/cookbook_spec.rb]: Not a fan of having multiple expectations in a single `it` block. Would it be possible to either split this up or do a single data structure comparison?

http://betterspecs.org/#single

Comment (kirtfitzpatrick) [inline spec/models/cookbook_spec.rb]: I was keeping with the style of the rest of the file:
https://github.com/chef/supermarket/blob/transfer_ownership/spec/models/cookbook_spec.rb

**Pull request merged**